File: databinding-base/src/main/java/com/hangox/databinding/recycleview/HBViewHolder.java
package com.hangox.databinding.recycleview;

import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * Created With Android Studio
 * User hangox
 * Date 15/12/28
 * Time 下午3:18
 */
public abstract class HBViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
    protected OnViewHolderClickListener mOnViewHolderClickListener;
    public HBViewHolder(View itemView) {
        super(itemView);

    }


    @Override
    public void onClick(View v) {
        if(mOnViewHolderClickListener != null) {
            mOnViewHolderClickListener.onRootViewClick(itemView,getAdapterPosition());
        }
    }

    public void setOnViewHolderClickListener(OnViewHolderClickListener onViewHolderClickListener) {
        itemView.setOnClickListener(onViewHolderClickListener == null ? null : this);
        mOnViewHolderClickListener = onViewHolderClickListener;
    }

    public interface OnViewHolderClickListener {
        void onRootViewClick(View rootView, int index);
    }
}

File: databinding-base/src/main/java/com/hangox/starter/recycleview/HBViewHolder.kt
package com.hangox.starter.recycleview

import android.view.View

import androidx.recyclerview.widget.RecyclerView

/**
 * Created With Android Studio
 * User hangox
 * Date 15/12/28
 * Time 下午3:18
 */
abstract class HBViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView),
    View.OnClickListener {
    protected var onViewHolderClickListener: OnViewHolderClickListener? = null
        set(value) {
            itemView.setOnClickListener(if (value == null) null else this)
            field = value
        }


    override fun onClick(v: View) {
        if (onViewHolderClickListener != null) {
            onViewHolderClickListener!!.onRootViewClick(itemView, adapterPosition)
        }
    }

    interface OnViewHolderClickListener {
        fun onRootViewClick(rootView: View, index: Int)
    }
}
